jira/src/getAuthType.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Returns if user is admin
+ * {boolean} true if user is current user is admin.
+ */
+function isAdminUser() {
+  return false;
+}
+
+/**
+ * Returns the Auth Type of this connector.
+ * @return {object} The Auth type.
+ */
+function getAuthType() {
+  var cc = DataStudioApp.createCommunityConnector();
+  return cc
+    .newAuthTypeResponse()
+    .setAuthType(cc.AuthType.USER_TOKEN)
+    .setHelpUrl('https://id.atlassian.com/manage/api-tokens')
+    .build();
+}
+
+/**
+ * Resets the auth service.
+ */
+function resetAuth() {
+  var user_tokenProperties = PropertiesService.getUserProperties();
+  user_tokenProperties.deleteProperty('dscc.username');
+  user_tokenProperties.deleteProperty('dscc.password');
+}
+
+/**
+ * Returns true if the auth service has access.
+ * @return {boolean} True if the auth service has access.
+ */
+function isAuthValid() {
+  var userProperties = PropertiesService.getUserProperties();
+  var userName = userProperties.getProperty('dscc.username');
+  var token = userProperties.getProperty('dscc.token');
+
+  return hasValue(userName) && hasValue(token);
```

Comment:
Should try a request to the Jira API here to make sure the provided credentials are valid.

Reply:
Thanks for the suggestion. That's what I wanted to do previously but I need the host to call the API which is being passed as a config parameter after the authentication, for this reason, I'm only validating fields to not be empty but will double-check to make sure.